### PR TITLE
Use SDK-style projects

### PR DESCRIPTION
**Pull request by sbomer**
This updates projects to use the new simplified format. ~~It depends on @jbevain's branch that I believe is a WIP: https://github.com/jbevain/cecil/tree/new-sdk.~~

~~This change will not be ready to merge until the cecil change is ready, but I wanted to share what it's going to look like and get early feedback.~~

Part of https://github.com/mono/linker/issues/452.

Edit: now that cecil has been updated to use SDK-style projects, this can move forward. Waiting on https://github.com/mono/linker/pull/489 to avoid merge conflicts in that PR.

### CODE REVIEW

**Comment by sbomer** [comment]
@marek-safar PTAL

File: external/Mono.Cecil.overrides [hunk 1]
```diff
@@ -1,13 +1,27 @@
 <Project>
 
+  <!-- Cecil's Directory.Build.Props imports this file if it
```
Comment (marek-safar) [inline external/Mono.Cecil.overrides]: Please don't modify the file here. It's submodule copy not actual location to make the changes

File: src/linker/Mono.Linker.csproj [hunk 2]
```diff
@@ -1,63 +1,38 @@
...
-    <SchemaVersion>2.0</SchemaVersion>
-    <ProjectGuid>{DD28E2B1-057B-4B4D-A04D-B2EBD9E76E46}</ProjectGuid>
     <OutputType>Exe</OutputType>
-    <AppDesignerFolder>Properties</AppDesignerFolder>
+    <EnableDefaultCompileItems>false</EnableDefaultCompileItems>
     <RootNamespace>Mono</RootNamespace>
-    <AssemblyName Condition=" ! $(ILLinkBuild) ">monolinker</AssemblyName>
-    <AssemblyName Condition=" $(ILLinkBuild) ">illink</AssemblyName>
-    <TargetFrameworkVersion>v4.6.2</TargetFrameworkVersion>
+    <Copyright>(C) 2006, Jb Evain</Copyright>
```
Comment (marek-safar) [inline src/linker/Mono.Linker.csproj]: Please use .NET Foundation for the copyright

**Comment by mrvoorhe** [comment]
@marek-safar @sbomer  Would it be OK if we hold off merging this PR until after the PR's that I have open are merged?  I suspect this change is going to cause me some issues I will have to work through.  Getting the improvement PR's landed and pulled into our UnityLinker is a higher priority for me than patching our projects back together.  If this PR lands first I'll have to start cherry picking around this change.  Not the end of the world, but if there's no rush it will make my life easier.

File: external/Mono.Cecil.overrides (hunk 1 above)
Comment (sbomer) [reply external/Mono.Cecil.overrides]: The contents of this file are purely to adapt the cecil project to our build (our directory layout, our use of arcade), which is the intended purpose of this import in cecil: https://github.com/jbevain/cecil/blob/master/Directory.Build.props#L24. I don't think any of these modifications belong in the cecil project.

File: src/linker/Mono.Linker.csproj (hunk 2 above)
Comment (sbomer) [reply src/linker/Mono.Linker.csproj]: I'm happy to change the copyright info, but I'd rather do it in a separate PR. Here my goal is just to use the new SDK while reproducing the original build as closely as possible.

**Comment by sbomer** [comment]
@mrvoorhe that's fine by me.

**Comment by sbomer** [comment]
@mrvoorhe is now a better time to make this change? This will unblock updates to cecil like https://github.com/mono/linker/pull/540.

**Comment by sbomer** [comment]
@marek-safar PTAL

**Comment by mrvoorhe** [comment]
Go for it.  Thanks for checking.

**Pull request merged**